lib/packages.cf

```diff
@@ -250,6 +250,22 @@ bundle common rpm_knowledge
       "rpm3_arch_regex" string => "\S+\s+(\S+).*";
 }
 
+bundle common redhat_no_locking_knowledge {
+  vars:
+    redhat_setopt_available::
+      "no_locking_option" string => "--setopt=exit_on_lock=True";
+    !redhat_setopt_available::
+      "no_locking_option" string => "";
+
+
+  classes:
+    # Use the class 'agent' as a guard, to prevent the command to be executed during cf-promises/cf-serverd
+    # Class agent is a hard class set by cf-agent
+    redhat.agent::
```

Comment:
Why did I not know about this `agent` class?

Of course, you can't see it from `cf-promises --show-classes` because that's not the agent.
But you can see it with the new `--show-evaluated-classes` option to `cf-agent`. 

```
> $ sudo cf-agent --show-evaluated-classes=^agent
Class name                                                   Meta tags                               
agent                                                        cfe_internal,source=agent,hardclass   
```


Thanks for the lesson!

Reply:
I discovered this option last week, when I detected that cf-promises ran the yum command.
I searched and digged, and found this class - it is very useful